testing/workflows/components/kfctl_go_test.jsonnet

```diff
@@ -0,0 +1,372 @@
+  activeDeadlineSeconds: 1800,  // Set 30 minute timeout for each template
+  workingDir: working_dir,
+  container: {
+    command: command,
+    image: image,
+    workingDir: working_dir,
+    // TODO(jlewi): Change to IfNotPresent.
+    imagePullPolicy: "Always",    
+    env: [
+      {
+        // Add the source directories to the python path.
+        name: "PYTHONPATH",
+        value: kubeflowPy + ":" + kubeflowTestingPy,
+      },
+      {
+        name: "GOOGLE_APPLICATION_CREDENTIALS",
+        value: "/secret/gcp-credentials/key.json",
+      },
+      {
+        name: "GITHUB_TOKEN",
+        valueFrom: {
+          secretKeyRef: {
+            name: "github-token",
+            key: "github_token",
+          },
+        },
+      },
+      {
+        // We use a directory in our NFS share to store our kube config.
+        // This way we can configure it on a single step and reuse it on subsequent steps.
+        // The directory should be unique for each workflow so that multiple workflows don't collide.
+        name: "KUBECONFIG",
+        value: kubeConfig,
+      },
+    ] + prowEnv + env_vars,
+    volumeMounts: [
+      {
+        name: dataVolume,
+        mountPath: mountPath,
+      },
+      {
+        name: "github-token",
+        mountPath: "/secret/github-token",
+      },
+      {
+        name: "gcp-credentials",
+        mountPath: "/secret/gcp-credentials",
+      },
+    ],
+  },
+  metadata: {
+    labels: prowDict {
+      workflow: params.name,
+      workflow_template: workflow_template,
+      step_name: step_name,
+    },
+  },
+  sidecars: sidecars,
+};  // buildTemplate
+
+local componentTests = util.kfTests {
+  name: "gke-tests",
+  platform: "gke",
+  testDir: testDir,
+  kubeConfig: kubeConfig,
+  image: image,
+  workflowName: params.workflowName,
+  buildTemplate+: {
+    argoTemplate+: {
+      container+: {
+        metadata+: {
+          labels: prowDict {
+            workflow: params.name,
+            workflow_template: workflow_template,
+          },
+        },
+      },
+    },
+  },
+};
+
+// Create a list of dictionary.c
+// Each item is a dictionary describing one step in the graph.
+local dagTemplates = [
+  {
+    template: buildTemplate("checkout",
+                            ["/usr/local/bin/checkout.sh", srcRootDir],
+                            env_vars=[{
+                              name: "EXTRA_REPOS",
+                              value: "kubeflow/tf-operator@HEAD;kubeflow/testing@HEAD",
+                            }]),
+    dependencies: null,
+  },  // checkout
+  {
+    template: buildTemplate("create-pr-symlink", [
+      "python",
+      "-m",
+      "kubeflow.testing.prow_artifacts",
+      "--artifacts_dir=" + outputDir,
+      "create_pr_symlink",
+      "--bucket=" + bucket,
+    ]),  // create-pr-symlink
+    dependencies: ["checkout"],
+  },  // create-pr-symlink
+  {
+    template: buildTemplate(
+      "build-kfctl",
+      [        
+        "pytest",
```

Comment:
IIRC invoke it with pytest the  __name__ == "__main__" will not be called.

Reply:
Correct. That's why we use options like --log-cli info. We rely on pytest to set log level appropriately to capture the stout and error.

The main function is just to make it easy to run in a debugger.